x-pack/plugins/viz_editor/public/frame/main/main.tsx

```diff
@@ -28,48 +32,66 @@ import { registry } from '../../editor_plugin_registry';
 type Action =
   | { type: 'loaded' }
   | { type: 'loadError'; message: string }
-  | { type: 'updateViewModel'; newState: ViewModel };
+  | { type: 'updateVisModel'; newState: VisModel };
+
+export interface MainProps {
+  getInterpreter: () => Promise<{ interpreter: any }>;
+  renderersRegistry: any;
+}
+
+export interface RootState {
+  visModel: VisModel;
+  // TODO stuff like dirt and valid will go in here
+  metadata: {};
+}
 
-function reducer(state: ViewModel, action: Action): ViewModel {
+function reducer(state: RootState, action: Action): RootState {
```

Comment:
Doesn't affect this PR, but it occurs to me that we're basically using this reducer as a `setState`, and I wonder if we even need to use reducers at all?

Reply:
At the moment we do, but for undo/redo we'll need the reducer (and also prob. for things like dirty state and invalid state) so I would keep it in place for now.